### DISCUSSION TITLE
Browser starts headless by default

### DISCUSSION THREAD

**Discussion by brssgrg** [seleniumbase/SeleniumBase#2416] [Q&A] [answered]
Hello,

i'm on Ubuntu 22 with GUI.
Why does this code not open a visible browser? I can see chrome process starting with ps and the code returns the cookie I want, but the browser is headless by default. I can see --headless=new in the process name.

```

def get_cf_cookie(url):
    
    with SB(uc=True) as sb:
        
        sb.open(url)
        sb.maximize_window()
        sb.sleep(3)
        
        cookies=sb.driver.get_cookies()

        for i in cookies:
            if i["name"] == "__cf_bm":
                cf_bm = i["value"]
    
    return cf_bm
```

**Comment by mdmintz** [comment, marked as answer]
You're looking for `headed=True`:

https://github.com/seleniumbase/SeleniumBase/blob/ba7f21a3e93e4fd742d29a107a9d84814207c553/seleniumbase/plugins/driver_manager.py#L69

Pass it as an arg to `SB()` or `Driver()`.

Otherwise Linux defaults to using headless mode.

**Reply by brssgrg** [reply]
Thank you, Michael!
